### PR TITLE
sink: The sink can make arbitrary HTTP requests

### PR DESCRIPTION
**Pull request by stefwalter**
This code change allows the sink to make arbitrary HTTP requests.
It always code do this, but was all framed within the GitHub
moniker. This is mostly cleanup.

### CODE REVIEW

File: sink/sink [hunk 1]
```diff
@@ -195,22 +195,23 @@ class GitHub(object):
 
     def push(self, status):
         github = status.get("github", { })
+        http = status.get("github", { })
```
Comment (martinpitt) [inline sink/sink]: I suppose you mean something else here, as otherwise this would just duplicate the entries in `requests`.

File: sink/sink [hunk 2]
```diff
@@ -195,22 +195,23 @@ class GitHub(object):
...
         token = github.get("token", self.token)
-        requests = github.get("requests", [])
+        requests = github.get("requests", []) + http.get("requests", [])
 
         for r in requests:
             self.results['link'] = status['link']
             result_name = r.get('result', None)
 
-            result = self.req(token, r)
+            result = self.req(r.get('token'), r)
```
Comment (martinpitt) [inline sink/sink]: Is there an `or token` missing, or will the request now always have the token due to the changes in `req()` above? The `token =` assingment from above is still being used further down in `self.watches(token, watches)`, does that not need to use the request specific token(s) as well?

File: sink/sink (hunk 1 above)
Comment (stefwalter) [reply sink/sink]: Thanks. Fixed.

File: sink/sink (hunk 2 above)
Comment (stefwalter) [reply sink/sink]: Thanks. Good catch. Fixed.

**Comment by stefwalter** [comment]
I rolled it out on fedorapeople.org, and it seems to be working since a few machines are still posting there.

**Pull request merged**